src/pages/MainPage/MainPage.tsx

```diff
@@ -1,7 +1,59 @@
-import React from 'react';
+import React, { useEffect } from 'react';
+import { GetNextPageParamFunction, InfiniteData, useInfiniteQuery } from '@tanstack/react-query';
+import { getPhotos } from 'api/unsplash-api';
+import { useInView } from 'react-intersection-observer';
+import { Photo } from 'components/Photo';
+import { ParamsFilter } from 'components/ui/ParamsFilter';
+import { Image, LikedImage, SearchResponse } from 'types/unsplash-types';
 
 export const pageTitle = 'MainPage';
 
 export const MainPage = () => {
-  return <div>{pageTitle}</div>;
+  const transformPhoto: (data: InfiniteData<SearchResponse | Image[], number>) => {
+    pages: LikedImage[][];
+    pageParams: number[];
+  } = (photo) => {
+    const photoWithLike = photo.pages.map((page) =>
+      ((page as SearchResponse).results || (page as Image[])).map((photo) => ({
+        ...photo,
+        isLiked: localStorage.getItem(String(photo.id)) === 'true',
+      }))
+    );
+
+    return {
+      pages: photoWithLike,
+      pageParams: photo.pageParams,
+    };
+  };
+
+  const getNextPageParam: GetNextPageParamFunction<number, SearchResponse | Image[]> = (
+    lastPage,
+    allPages,
+    lastPageParam
+  ) => {
+    if (((lastPage as SearchResponse).results || (lastPage as Image[])).length === 0) return;
+    return lastPageParam + 1;
+  };
+
+  const { ref, inView, entry } = useInView();
+  const { data, fetchNextPage, isFetchingNextPage, hasNextPage } = useInfiniteQuery({
+    queryKey: ['photos'],
```

Comment:
u should add filter params in queryKey array, when u add filter here u don't need to use invalidateQueries from hook because useQuery will call api after updating filter